execution/engine.go

```diff
@@ -280,8 +280,29 @@ func (e *Engine) SubmitMarketWithLiquidityProvision(ctx context.Context, marketC
 	lpID,
 	deterministicId string,
 ) error {
+	return e.submitOrRestoreMarketWithLiquidityProvision(ctx, marketConfig, lp, party, lpID, deterministicId, true)
+}
+
+// RestoreMarketWithLiquidityProvision is very similar to SubmitMarketWithLiquidityProvision without updating the market tracker as it is restored separately from checkpoint.
+func (e *Engine) RestoreMarketWithLiquidityProvision(ctx context.Context, marketConfig *types.Market, lp *types.LiquidityProvisionSubmission, lpID, deterministicId string) error {
+	proposer := e.marketActivityTracker.GetProposer(marketConfig.ID)
+	if len(proposer) == 0 {
+		return ErrMarketDoesNotExist
+	}
+	return e.submitOrRestoreMarketWithLiquidityProvision(ctx, marketConfig, lp, proposer, lpID, deterministicId, false)
```

Comment:
Will it be fine here to use the market ID as the proposer of the market? will we not be submitting the LP for a wrong party ID?

Reply:
we're not submitting the market ID, we're submitting the proposer of the market as the proposer.

Reply:
oh right. I'm an idot indeed.